reconcile/test/test_terraform_cloudflare_resources.py

```diff
@@ -269,3 +358,51 @@ def test_cloudflare_namespace_validation(
     assert ["No cloudflare namespaces were detected, nothing to do."] == [
         rec.message for rec in caplog.records
     ]
+
+
+def custom_ssl_secret_reader_side_effect(*args):
+    """For use of secret_reader inside cloudflare client"""
+    if {
+        "path": "certificate/secret/cert/path",
+        "field": "certificate.crt",
+        "version": 1,
+        "q_format": "plain",
+    } == args[0]:
+        return "----- CERTIFICATE -----"
+
+    if {
+        "path": "certificate/secret/cert/path",
+        "field": "certificate.key",
+        "version": 1,
+        "q_format": "plain",
+    } == args[0]:
+        return "----- KEY -----"
+
+
+def test_terraform_cloudflare_resources_dry_run(
```

Comment:
From the test implementation it looks like the purpose of this test is to verify we don't run terraform apply. In that case, you don't need any specific implementation for custom ssl resource in `custom_ssl_secret_reader_side_effect`. 

Also take a look at `test_terrascript_cloudflare_client.py` , this is where we actually test the terraform resource generation. You can add a test there for `cloudflare_custom_ssl` resource and use mocking for secret reader there with the side effect logic you have  in `custom_ssl_secret_reader_side_effect`

Reply:
just found the similar example here https://github.com/app-sre/qontract-reconcile/blob/master/reconcile/test/utils/test_terrascript_cloudflare_client.py#L433

Reply:
I've copied this pattern from https://github.com/app-sre/qontract-reconcile/blob/master/reconcile/test/test_terraform_cloudflare_users.py#L465

which is similar in use and behavior. I was to lazy to create the `expected_args` because the resource definition for this one was way bigger.

Also to remove the `custom_ssl_secret_reader_side_effect` I would need to remove the `custom_ssl` from the resources mocked as well.

Reply:
ah right, makes sense!

IMO, you can also add a test in `test_terrascript_cloudflare_client` to actually verify terraform resource structure. That will give us more confident that we generate the right resource.

Reply:
Done.